libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -743,9 +755,15 @@ func (dhtPeer *DHTPeer) handleNewDelegationConnection(conn net.Conn) {
 	dhtPeer.agentRecordsLock.Lock()
 	dhtPeer.agentRecords[addr] = record
 	dhtPeer.agentRecordsLock.Unlock()
+
 	dhtPeer.tcpAddressesLock.Lock()
 	dhtPeer.tcpAddresses[addr] = conn
 	dhtPeer.tcpAddressesLock.Unlock()
+
+	dhtPeer.acnStatusesLock.Lock()
+	dhtPeer.acnStatuses[addr] = make(chan *acn.Status, 1000)
+	dhtPeer.acnStatusesLock.Unlock()
```

Comment:
for the future... we could wrap a map in a struct to handle the lock and unlock, similar to:

https://github.com/orcaman/concurrent-map

Reply:
i suppose we can even use one lock for all structure, but need to check